### PR TITLE
Port CI to GitHub actions

### PR DESCRIPTION
**Pull request by daniele-moro**

### CODE REVIEW

**Comment by daniele-moro** [comment]
@EmanueleGallone @ccascone

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+          load: true # load the builded image to local docker instance. Needed for testing step.
+          cache-from: "${{ env.DOCKER_REPO }}:${{ matrix.TAGNAME }}"
+          build-args: |
+            "JOBS=${{ env.JOBS }}"
+            "GRPC_VER=${{ env.GRPC_VER }}"
+            "PI_COMMIT=${{ matrix.PI_COMMIT }}"
+            "BMV2_COMMIT=${{ matrix.BMV2_COMMIT }}"
+            "PI_CONFIGURE_FLAGS=${{ env.PI_CONFIGURE_FLAGS }}"
+            "BMV2_CONFIGURE_FLAGS=${{ matrix.BMV2_CONFIGURE_FLAGS }}"
+          tags: "${{ env.DOCKER_REPO }}:p4c-${{ matrix.TAGNAME }}" # Local Runtime image
```
Comment (EmanueleGallone) [inline .github/workflows/main.yaml:82]: I know this is an intermediate image, but maybe use another tag? using `p4c` could be misleading
```suggestion
          tags: "${{ env.DOCKER_REPO }}:p4mn-${{ matrix.TAGNAME }}" # Local Runtime image
```

File: .github/workflows/main.yaml [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+          target: "runtime"
+
+      - name: Test docker build
+        run: |
+          docker run --rm ${{ env.DOCKER_REPO }}:p4c-${{ matrix.TAGNAME }} simple_switch_grpc--help
```
Comment (EmanueleGallone) [inline .github/workflows/main.yaml:87]: Also here

File: .github/workflows/main.yaml [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+          docker run --rm ${{ env.DOCKER_REPO }}:p4c-${{ matrix.TAGNAME }} mn --help
```
Comment (EmanueleGallone) [inline .github/workflows/main.yaml:88]: And here

**Review by EmanueleGallone — DISMISSED** [review]
LGTM, minor comment

**Review by ccascone — APPROVED** [review]

**Comment by daniele-moro** [comment]
test this please

**Comment by daniele-moro** [comment]
retest this please

**Pull request merged**